sdk/core/azure-core/inc/azure/core/operation.hpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "azure/core/context.hpp"
+#include "azure/core/operation_status.hpp"
+#include "azure/core/response.hpp"
+
+#include <chrono>
+#include <string>
+#include <memory>
+
+namespace Azure { namespace Core {
+
+  //@param <T> The type of the final result of the long-running operation.
+  template <class T> class Operation {
+  private:
+    virtual std::unique_ptr<Http::RawResponse> PollInternal(Context& context) = 0;
+    virtual Response<T> PollUntilDoneInternal(std::chrono::milliseconds period, Context& context)
```

Comment:
That is the dotnet guidelines not ours.

That said the testcase displays how it would be used and eventually the storage team will add usage patterns for each Operation

Reply:
> That is the dotnet guidelines not ours.

I think the API design principle holds though :)

> That said the testcase displays how it would be used and eventually the storage team will add usage patterns for each Operation

Sounds good.